packages/base/tests/deserialize.test.ts

```diff
@@ -0,0 +1,319 @@
+import JSBI from "jsbi";
+import { HashType } from "@ckb-lumos/base";
+import { DepType } from "./../src/api.d";
+import test from "ava";
+import * as blockchain from "../src/blockchain";
+import type * as api from "../src/api";
+import { BI } from "@ckb-lumos/bi";
+
+test.before(() => {
+  JSBI.prototype["valueOf"] = () => 0;
+});
```

Comment:
any reason for the overriding here?

Reply:
The behaviour of `deepEqual` in ava test environment is to recursively call `valueOf` of every field in the values to be compared. 

Since `BI` in `@ckb-lumos/bi` has a `JSBI` instance, `jsbi.valueOf` will be called when calling `t.deepEqual(BI.from(1), BI.from(1))`, and that will always throw an Error: https://github.com/GoogleChromeLabs/jsbi/blob/main/lib/jsbi.ts#L80

Reply:
i see, but the `JSBI.prototype["valueOf"] = () => 0;` may cause all `t.deepEquals` to be true when comparing `JSBI`/`BI`. It would be better to use `JSBI.prototype.valueOf = function() { return this.toString(); }`

Reply:
Also, I would say that this code is confusing, and we should comment the reason for doing this